test/logflare_web/controllers/api/team_controller_test.exs

```diff
@@ -0,0 +1,171 @@
+defmodule LogflareWeb.Api.TeamControllerTest do
+  use LogflareWeb.ConnCase
+
+  import Logflare.Factory
+
+  alias Logflare.Sources.Counters
+
+  setup do
+    insert(:plan, name: "Free")
+    user = insert(:user)
+    main_team = insert(:team, user: user)
+
+    another_user_team_with_main_user = insert(:team, user: insert(:user))
+    insert(:team_user, team: another_user_team_with_main_user, email: user.email)
+
+    _non_relevant_to_main_user = insert(:team_user, team: main_team, email: insert(:user).email)
+
+    Counters.start_link()
+
+    {:ok, user: user, main_team: main_team, non_owner_team: another_user_team_with_main_user}
+  end
+
+  describe "index/2" do
+    test "returns list of teams for given user", %{
+      conn: conn,
+      user: user,
+      main_team: main_team,
+      non_owner_team: non_owner_team
+    } do
+      response =
+        conn
+        |> login_user(user)
+        |> get("/api/teams")
+        |> json_response(200)
+
+      response = response |> Enum.map(& &1["name"]) |> Enum.sort()
+      expected = [main_team, non_owner_team] |> Enum.map(& &1.name) |> Enum.sort()
```

Comment:
we can do this instead to make it more concise.
```
assert [_, _] = json_resonse(conn, 200)
```
don't need to check equality, since we know there will only be 2 teams.

Reply:
there are more teams created during setup ( the `_non_relevant_to_main_user`) to ensure we're fetching with authorization in mind hence the check of equality

Reply:
i see. fetching a two len array would imply that it is working correctly as well.

if an equality check is really needed, would be simpler to check if name is in the response name array, as this is rather hard to read.

Reply:
oh good call! I can even get by token because I've added the token to the jason derive